src/quantilealgs.jl

```diff
@@ -1,16 +1,25 @@
 # Various algorithms for computing quantile
 
-function quantile_bisect(
-    d::ContinuousUnivariateDistribution, p::Real, lx::Real, rx::Real,
+function quantile_bisect(d::ContinuousUnivariateDistribution, p::Real, lx::T, rx::T) where {T<:Real}
+    rx < lx && throw(ArgumentError("[$lx, $rx] is not a valid bracketing interval"))
+
+    # In some special cases, e.g. #1501, rx == lx`
+    # If the distribution is degenerate the check below can fail, hence we skip it
+    if rx == lx
+        # Returns `lx` of the same type as `(lx + rx) / 2`
+        # For specific types such as `Float64` it is more performant than `oftype((lx + rx) / 2, lx)`
+        return middle(lx)
+    end
+
     # base tolerance on types to support e.g. `Float32` (avoids an infinite loop)
     # ≈ 3.7e-11 for Float64
     # ≈ 2.4e-5 for Float32
-    tol::Real=(eps(Base.promote_typeof(float(lx), float(rx))))^(2 / 3),
-)
+    tol = eps(float(T))^(2 / 3)
```

Comment:
That widens `Float32` to `Float64`, but this seems okay

Reply:
Ah true :+1: Since the line is changed anyway it might be a good opportunity to fix it:
```suggestion
    tol = cbrt(eps(float(T)))^2
```